### PR TITLE
TEL-4801 add custom alert type based on logs as a percentage

### PR DESCRIPTION
**Pull request by sjpalf**
What did we do?
--

1. Added new custom alert type for alerts based on logs as a percentage

References
--

https://jira.tools.tax.service.gov.uk/browse/TEL-4801

Evidence of work
--

1. Tested from branch:
image

Next Steps
--

1. Add api 5xx alert in alert-config
2. Add new module in telemetry-grafana-alert-config

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/alertconfig/builder/custom/CustomElasticsearchPercentageAlert.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  * @param summary
+  *   The description to populate in PagerDuty when the alert fires
+  * @param teamName
+  *   All alerts are prefixed with the team name
+  * @param thresholds
+  *   Trigger point for each environment
+  */
+case class CustomElasticsearchPercentageAlert(
+    alertName: String,
+    checkIntervalMinutes: CheckIntervalMinutes = CheckIntervalMinutes.TWO_MINUTES,
```
Comment (ma3574) [inline src/main/scala/uk/gov/hmrc/alertconfig/builder/custom/CustomElasticsearchPercentageAlert.scala:58]:
```suggestion
    checkIntervalMinutes: Option[CheckIntervalMinutes] = None,
```
We do it this way in the other alert types. We came up with a rule that we only send things in the YAML that are non default. The deafulting happens in the Terraform. I don't know if it's the best way but for consistency.

File: src/test/scala/uk/gov/hmrc/alertconfig/builder/AlertConfigBuilderSpec.scala [hunk 2]
```diff
@@ -19,7 +19,24 @@ package uk.gov.hmrc.alertconfig.builder
...
+  YamlHttp5xxPercentThresholdAlert,
+  YamlHttp5xxThresholdAlert,
+  YamlHttpAbsolutePercentSplitDownstreamHodThresholdAlert,
+  YamlHttpAbsolutePercentSplitDownstreamServiceThresholdAlert,
+  YamlHttpAbsolutePercentSplitThresholdAlert,
+  YamlHttpStatusPercentThresholdAlert,
+  YamlHttpStatusThresholdAlert,
+  YamlHttpTrafficThresholdAlert,
+  YamlTotalHttpRequestThresholdAlert
+}
```
Comment (ma3574) [inline src/test/scala/uk/gov/hmrc/alertconfig/builder/AlertConfigBuilderSpec.scala:39]:
```suggestion
import uk.gov.hmrc.alertconfig.builder.yaml._
```

File: src/main/scala/uk/gov/hmrc/alertconfig/builder/custom/CustomElasticsearchPercentageAlert.scala (hunk 1 above)
Comment (sjpalf) [reply src/main/scala/uk/gov/hmrc/alertconfig/builder/custom/CustomElasticsearchPercentageAlert.scala:58]: fixed

**Review by ma3574 — APPROVED** [review]

**Pull request merged**